optuna_dashboard/ts/components/GraphTimeline.tsx

```diff
@@ -65,13 +66,34 @@ const plotTimeline = (trials: Trial[], mode: string) => {
       )
     )
   )
-  const maxDatetime = new Date(
-    Math.max(
-      ...lastTrials.map(
-        (t) => t.datetime_start?.getTime() ?? minDatetime.getTime()
-      )
-    )
+  const maxRunDuration = Math.max(
+    ...trials.map((t) => {
+      return t.datetime_start === undefined || t.datetime_complete === undefined
+        ? -Infinity
+        : t.datetime_complete.getTime() - t.datetime_start.getTime()
+    })
   )
+  const hasRunning =
+    (maxRunDuration === -Infinity &&
+      trials.some((t) => t.state === runningKey)) ||
+    trials.some((t) => {
+      if (t.state !== runningKey) {
+        return false
+      }
+      const start = t.datetime_start?.getTime() ?? new Date().getTime()
+      const now = new Date().getTime()
+      // This is an ad-hoc handling to check if the trial is running.
+      return now - start < maxRunDuration * 5
```

Comment:
[question] Why do we need those lines for ad-hoc handling?

Reply:
For me, it can be just simply with `  const hasRunning = trials.some((t) => t.state === runningKey)`, but do you have some reasons that you need to write those lines?

Reply:
We can remove it as well, but then the problem is that the timeline for `RUNNING` trials, which was not somehow killed properly (it is basically preemption, which often happens, in our internal cluster) then the timeline plot for the other trials will not be visible because they will be squashed by the running trials, which are not really running because of preemption.